source-code/core/src/config/zod.ts

```diff
@@ -10,12 +9,15 @@ import { Resource } from "../ast/zod.js"
  * at https://zod.dev/
  */
 export const Config = z.object({
-	referenceLanguage: z.string().transform((value) => value as Language),
-	languages: z.array(z.string()),
+	referenceLanguage: z.string(),
+	languages: z.array(z.string()).refine((items) => new Set(items).size === items.length, {
```

Comment:
> what if we use preprocess to remove duplicates instead of throwing an error? 

Implicit behavior. If a plugin adds the same language multiple times, something is wrong with the plugin. Plugin authors should address that problem instead of us implicitly "fixing" the error but not throwing but maybe causing (and not catching) later-stage crashes.

Reply:
Wups, `pluginSetup` removes duplicates, see 

https://github.com/inlang/inlang/blob/ec15170bc17ef36916123152d2fdbc3b2e8c0d91/source-code/core/src/plugin/setupPlugins.ts#L51-L52. 

The reasoning is though that multiple plugins might add languages in the setup process because they are unaware of each other. Slightly different reasoning to the above of a standalone plugin defining duplicates languages

Reply:
> If a plugin adds the same language multiple times, something is wrong with the plugin. Plugin authors should address that problem instead of us implicitly "fixing" the error but not throwing and thereby maybe causing (and not catching) later-stage crashes.

But it will affect end users and they will need to wait a couple of weeks until the plugin author fixes the issue..

Reply:
I don't think this situation will arise. The plugin template provides out of the box tests. With the introduction of the plugin API and new `@inlang-git/fs` even in a single line of code. 

I propose to wait and see if our plugin tests prevent situations in which plugins break.